0004-post-tags.md

```diff
@@ -0,0 +1,299 @@
+- Feature Name: Post Tags
+- Start Date: 2023-11-04
+- RFC PR: [LemmyNet/rfcs#0004](https://github.com/LemmyNet/rfcs/pull/4)
+- Lemmy Issue: [LemmyNet/lemmy#317](https://github.com/LemmyNet/lemmy/issues/317)
+
+# Summary
+
+Ability to add descriptive tags to Posts similar to Reddit's Flair system primarily on a per-community basis.
+
+# Motivation
+
+Post Tags would allow for vastly improved content filtering (and to a lesser extent searching) both for Users and Admins/Moderators. 
+
+# Guide-level explanation
+
+Tags grant you more fine grained control over the content you see on Lemmy. Certain users do not want to see certain content and tags allow you to do exactly that.
+
+For example a lot of communities offer mixed content such as Memes, News and Discussions. Some people would like to only see News or Discussions. In such a case they could blacklist (filter out) the "Meme" tag, going forward they would not be presented with any Memes in that community.
```

Comment:
You'd need to clarify then, who is creating and applying these tags. Because if its the posters / users themselves, rather than the community moderators, then it won't help you with filtering.

If the tags are completely unmoderated, and anyone can tag something with any tag they want, it would be far worse for that than subscribing or blocking as it exists already.

Reply:
A potential "curator" moderation level whose entire purpose is to adjust tags could be introduced, though you could definitely make the argument that this is overkill for most kinds of communities.

Reply:
Intended curation for the start is that Admins/Moderators create these tags, mostly because Lemmy's current permission system doesn't allow for much anything else without significant work. Clarified the language in the section a bit (see d894b357ef2372ec56a9b3e75fa95bc5d315ac94).